File: src/main/scala/com/hoi4utils/clausewitz/script/ScopedPDXScript.scala
package com.hoi4utils.clausewitz.script

import com.hoi4utils.clausewitz.code.scope.{Scope, ScopeType}

trait ScopedPDXScript {
  def definitionScope: Option[Scope]

  def isSupportedInScope(scope: Scope): Boolean = {
    if (supportedScopes.exists(_ == ScopeType.any)) return true
    supportedScopes.exists(_ == scope.targetScopeType)
  }

  def supportedScopes: Iterable[ScopeType]
}

File: src/main/java/com/hoi4utils/clausewitz/script/ScopedPDXScript.java
//package com.hoi4utils.clausewitz.script;
//
//import com.hoi4utils.clausewitz.code.scope.Scope;
//import com.hoi4utils.clausewitz.code.scope.ScopeType;
//
//import java.util.Collection;
//
//public interface ScopedPDXScript {
//    Scope getDefinitionScope();
//
//    default boolean isSupportedInScope(Scope scope) {
//        if (supportedScopes().contains(ScopeType.any))
//            return true;
//        return supportedScopes().contains(scope.targetScopeType());
//    }
//
//    Collection<ScopeType> supportedScopes();
//}
